workers/generic-worker/loopback_video.go

```diff
@@ -0,0 +1,85 @@
+//go:build darwin || linux
+
+// TODO: are these right? ^^
+// I was having issues with them only being set to linux
+
+package main
+
+import (
+	"fmt"
+
+	"github.com/taskcluster/taskcluster/v53/internal/scopes"
+	"github.com/taskcluster/taskcluster/v53/workers/generic-worker/host"
+)
+
+type LoopbackVideoFeature struct {
+}
+
+func (feature *LoopbackVideoFeature) Name() string {
+	return "LoopbackVideo"
+}
+
+func (feature *LoopbackVideoFeature) Initialise() error {
+	return nil
+}
+
+func (feature *LoopbackVideoFeature) PersistState() error {
+	return nil
+}
+
+func (feature *LoopbackVideoFeature) IsEnabled(task *TaskRun) bool {
+	return task.Payload.Features.LoopbackVideo
+}
+
+func (feature *LoopbackVideoFeature) NewTaskFeature(task *TaskRun) TaskFeature {
+	return &LoopbackVideoTask{
+		task: task,
+	}
+}
+
+type LoopbackVideoTask struct {
+	task *TaskRun
+}
+
+func (lvt *LoopbackVideoTask) RequiredScopes() scopes.Required {
+	return scopes.Required{
+		// TODO: should we keep as the same format as docker worker?
+		// would make the translation easier in d2g, but would not
+		// be impossible to translate if we used something like:
+		// generic-worker:features:loopbackVideo
+		{"generic-worker:capability:device:loopbackVideo"},
+		{fmt.Sprintf("generic-worker:capability:device:loopbackVideo:%s", lvt.task.Definition.TaskQueueID)},
+		// TODO: is this one necessary?
+		{"generic-worker:capability:device:loopbackVideo", fmt.Sprintf("generic-worker:capability:device:loopbackVideo:%s", lvt.task.Definition.TaskQueueID)},
+	}
+}
+
+func (lvt *LoopbackVideoTask) ReservedArtifacts() []string {
+	return []string{}
+}
+
+func (lvt *LoopbackVideoTask) Start() *CommandExecutionError {
+	err := host.Run("/bin/bash", "-c", "/usr/bin/sudo modprobe v4l2loopback video_nr=0")
+	if err != nil {
+		return executionError(internalError, errored, fmt.Errorf("Could not load the v4l2loopback kernel module: %v", err))
+	}
+
+	err = host.Run("/bin/bash", "-c", "/usr/bin/sudo chmod 660 /dev/video0")
+	if err != nil {
+		return executionError(internalError, errored, fmt.Errorf("Could not chmod 660 the /dev/video0 device: %v", err))
+	}
+
+	err = makeFileOrDirReadWritableForUser(false, "/dev/video0", taskContext.User)
```

Comment:
Not sure if taskContext is initialised at this point. Hopefully! I just can't remember. My own fault - having taskContext as a global variable was a pretty ugly design decision of mine.

Reply:
Looks like it's set beforehand :) https://community-tc.services.mozilla.com/tasks/CmLsFsBDTzeCVgaDX4EasA/runs/0/logs/public/logs/live.log#L191